metrics/metricstest/resource_metrics.go

```diff
@@ -298,25 +328,27 @@ func (v Value) Equal(other Value) bool {
 		if v.Distribution.Count != other.Distribution.Count {
 			return false
 		}
-		if v.Distribution.Sum != other.Distribution.Sum {
-			return false
-		}
-		if v.Distribution.SumOfSquaredDeviation != other.Distribution.SumOfSquaredDeviation {
-			return false
-		}
-		if v.Distribution.BucketOptions != nil {
-			if other.Distribution.BucketOptions == nil {
+		if !v.VerifyDistributionCountOnly && !other.VerifyDistributionCountOnly {
```

Comment:
I'd use an early-exit here, rather than adding an extra level of nesting.

```golang
if v.VerifyDistributionCountOnly || other.VerifyDistributionCountOnly {
  return true
}
...
```

Reply:
Done.